packages/retail-ui-extensions/src/extension-api/scanner-api/scanner-api.ts

```diff
@@ -0,0 +1,32 @@
+import type {RemoteSubscribable} from '@remote-ui/async-subscription';
+
+/** The scanner source the POS device supports. */
+export type ScannerSource = 'camera' | 'external' | 'embedded';
+
+export interface ScannerSubscriptionResult {
+  /** The string data from the last scanner event received. */
+  data: string;
```

Comment:
i think data and source might have to be undefined, im getting problems with the extension running if i destructure these from the useScannerDataSubscription..

Reply:
Yeah it marks the whole api as undefined causing the extension to not load. You will have to do something like this   `const { data, source } = useScannerDataSubscription() || {};` which isn't ideal for the developer experience.

Reply:
We paired on this, what Victor showed makes sense. I agree that making those optional is not ideal but creates a better development experience.